Build issue

Hi, 
I have downloaded pyql and Quantlib to a new Linux Mint system ( Linux Mint 17.2 cinnamon 64bit). 
I am unable to build pyql with errors similar to [103]( https://github.com/enthought/pyql/pull/103). 
The quantlib has build fine and I can run the examples fine. Please also note i am using Cython 0.23.4. 
Any help is much appreciated,  the error is attached below : 

x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -fPIC -I/usr/local/include -I/usr/include -I. -I./cpp_layer -I/usr/lib/python2.7/dist-packages/numpy/core/include -I/usr/include/python2.7 -c quantlib/time/date.cpp -o build/temp.linux-x86_64-2.7/quantlib/time/date.o
quantlib/time/date.cpp: In function ‘QuantLib::TimeUnit __Pyx_PyInt_As_enum__QuantLib_3a__3a_TimeUnit(PyObject*)’:
quantlib/time/date.cpp:11386:188: error: invalid cast from type ‘QuantLib::Period’ to type ‘QuantLib::TimeUnit’
                             return (enum QuantLib::TimeUnit) (((enum QuantLib::TimeUnit)-1)*(((((enum QuantLib::TimeUnit)digits[1]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[0])));
                                                                                                                                                                                            ^
quantlib/time/date.cpp:11404:245: error: invalid cast from type ‘QuantLib::Period’ to type ‘QuantLib::TimeUnit’
                             return (enum QuantLib::TimeUnit) (((enum QuantLib::TimeUnit)-1)*(((((((enum QuantLib::TimeUnit)digits[2]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[1]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[0])));
                                                                                                                                                                                                                                                     ^
quantlib/time/date.cpp:11422:302: error: invalid cast from type ‘QuantLib::Period’ to type ‘QuantLib::TimeUnit’
                             return (enum QuantLib::TimeUnit) (((enum QuantLib::TimeUnit)-1)*(((((((((enum QuantLib::TimeUnit)digits[3]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[2]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[1]) << PyLong_SHIFT) | (enum QuantLib::TimeUnit)digits[0])));

--- a/quantlib/time/_period.pxd
+++ b/quantlib/time/_period.pxd
@@ -19,7 +19,7 @@ cdef extern from 'ql/time/frequency.hpp' namespace "QuantLib":
 
 cdef extern from 'ql/time/timeunit.hpp' namespace "QuantLib":
     cdef enum TimeUnit:
-        Days
+        Days,
         Weeks,
         Months,
         Years

--- a/quantlib/time/date.pyx
+++ b/quantlib/time/date.pyx
@@ -133,13 +133,13 @@ cdef class Period:
                                      % tenor)
                 self._thisptr = \
                 new shared_ptr[QlPeriod](new QlPeriod(<Integer> int(t),
-                                         <_period.TimeUnit> _TU_DICT[u]))
+                                         <TimeUnit> _TU_DICT[u]))
             else:
                 self._thisptr = \
                 new shared_ptr[QlPeriod](new QlPeriod(<_period.Frequency>args[0]))
         elif len(args) == 2:
             self._thisptr = new shared_ptr[QlPeriod](
-                new QlPeriod(<Integer> args[0], <_period.TimeUnit> args[1]))
+                new QlPeriod(<Integer> args[0], <TimeUnit> args[1]))
         elif len(args) == 0:
             self._thisptr = new shared_ptr[QlPeriod]()
         else:

